### PR TITLE
rusk: derive previous provisioners using current state

### LINKED ISSUE #1516: Avoid having to perform two calls to the staking contract when loading a state
#### Summary
At the moment when loading a state we need to get the new state and the old state. See https://github.com/dusk-network/rusk/pull/1513#discussion_r1516380001.

#### Possible solution design or implementation
Track the last block changes in the stake contract so that the node is able can retrieve the previous state without performing any additional calls to get previous provisioners. It needs to be said that this is something we need because of the consensus. Tracking state is not strictly necessary for 3rd party contracts.

#### Additional context
This is urgent because we need to implement the state snapshot for the hyperfast sync-up without resorting to multiple calls

### PR DESCRIPTION
**Pull request by herr-seppia**
Resolves #1516

Unblock #1094 and #1513

### CODE REVIEW

**Review by ureeves — APPROVED** [review]
Very nice! No more extra states. With the speedup in getting provisioners this should be no problem!

**Comment by herr-seppia** [comment]
Get_provisioners is no longer used to track changes since a while.
This PR just remove the need to get previous state provisioners during startup/fallback

File: node/src/chain/fallback.rs [hunk 1]
```diff
@@ -99,15 +92,25 @@ impl<'a, N: Network, DB: database::DB, VM: vm::VMExecution>
...
             .vm
             .read()
             .await
-            .get_provisioners(prev_prev_header.state_hash)?;
-
-        let mut provisioners_list = ContextProvisioners::new(provisioners_list);
-        provisioners_list.set_previous(prev_provisioners_list);
+            .get_changed_provisioners(prev_header.state_hash)?;
+
+        if !changed_provisioners.is_empty() {
```
Comment (goshawk-3) [inline node/src/chain/fallback.rs:104]: nitpick: As this is duplicated code, can we put it into a function for easier maintenance?

Comment (herr-seppia) [reply node/src/chain/fallback.rs]: Done

**Review by ureeves — APPROVED** [review]

**Review by goshawk-3 — APPROVED** [review]

**Pull request merged**